lib/install/management/configure.go

```diff
@@ -98,50 +98,65 @@ func (d *Dispatcher) Configure(vch *vm.VirtualMachine, conf *config.VirtualConta
 
 	snapshotName := fmt.Sprintf("%s %s", ConfigurePrefix, conf.Version.BuildNumber)
 	snapshotName = strings.TrimSpace(snapshotName)
-
 	// check for old snapshot
-	// #nosec: Errors unhandled.
-	oldSnapshot, _ := d.appliance.GetCurrentSnapshotTree(d.op)
+	oldSnapshot, err := d.appliance.GetCurrentSnapshotTree(d.op)
+	if err != nil {
+		// log the error but continue
+		d.op.Debugf("Error checking appliance snapshot tree during %s: %s", d.Action.String(), err.Error())
+	}
 
 	newSnapshotRef, err := d.createSnapshot(snapshotName, "configure snapshot")
 	if err != nil {
-		d.deleteUpgradeImages(ds, settings)
+		d.deleteISOs(ds, settings)
 		return err
 	}
 
-	err = d.update(conf, settings, isConfigureOp)
+	// rollback function
+	rollback := func() {
```

Comment:
I'm not sure I love this rollback function pattern. I don't love the pattern that was here before either, but this is still really fragmented. For example, we still have a one-off `defer` above to do one piece of the rollback and now it seems like we leave the snapshot behind when we rollback (instead of rolling back and then deleting it) since we `return err` after `rollback()`.